### PR TITLE
246 extra pnc search fields

### PR DESCRIPTION
**Pull request by retroredge**

### CODE REVIEW

File: app/services/helpers/SearchQueryBuilder.java [hunk 1]
```diff
@@ -75,22 +82,34 @@ public static SearchSourceBuilder searchSourceFor(String searchTerm, int pageSiz
...
+            .map(term -> covertToCanonicalDate(term).get())
             .collect(toList());
     }
 
-    public static String termsWithoutDates(String searchTerm) {
+    public static String simpleTerms(String searchTerm) {
         return Stream.of(searchTerm.split(" "))
-            .filter(term -> !DateTimeHelper.covertToCanonicalDate(term).isPresent())
+            .filter(term -> term.length() > 1)
+            .filter(term -> !covertToCanonicalDate(term).isPresent())
```
Comment (andymarke) [inline app/services/helpers/SearchQueryBuilder.java]: use `not()`

File: test/helpers/PncHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        assertThat(canBeConvertedToAPnc("03/4567A")).isTrue();
+        assertThat(canBeConvertedToAPnc("03/567A")).isTrue();
+        assertThat(canBeConvertedToAPnc("03/67A")).isTrue();
+        assertThat(canBeConvertedToAPnc("03/7A")).isTrue();
+        assertThat(canBeConvertedToAPnc("03/7A")).isTrue();
+    }
+
+    @Test
+    public void rejectsNonPncFormattedTerms() {
+        assertThat(canBeConvertedToAPnc("2003/A")).isFalse();
```
Comment (andymarke) [inline test/helpers/PncHelperTest.java]: Empty String test

File: test/helpers/PncHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        assertThat(canBeConvertedToAPnc("203/1234567A")).isFalse();
+        assertThat(canBeConvertedToAPnc("1234567A")).isFalse();
+        assertThat(canBeConvertedToAPnc("2013")).isFalse();
+        assertThat(canBeConvertedToAPnc("john smith")).isFalse();
+        assertThat(canBeConvertedToAPnc("16/11/2018")).isFalse();
+        assertThat(canBeConvertedToAPnc("111111/11A")).isFalse();
+        assertThat(canBeConvertedToAPnc("SF68/945674U")).isFalse();
```
Comment (andymarke) [inline test/helpers/PncHelperTest.java]: test andy/marke

File: app/services/helpers/SearchQueryBuilder.java (hunk 1 above)
Comment (retroredge) [reply app/services/helpers/SearchQueryBuilder.java]: Done

File: app/helpers/PncHelper.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package helpers;
```
[the remark this answers is not in the document]
Comment (retroredge) [reply app/helpers/PncHelper.java]: Done

File: app/helpers/PncHelper.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+public class PncHelper {
```
[the remark this answers is not in the document]
Comment (retroredge) [reply app/helpers/PncHelper.java]: Done

File: test/helpers/PncHelperTest.java (hunk 2 above)
Comment (retroredge) [reply test/helpers/PncHelperTest.java]: Done

File: test/helpers/PncHelperTest.java (hunk 3 above)
Comment (retroredge) [reply test/helpers/PncHelperTest.java]: Done

**Pull request merged**